expenses-manager-frontend/src/main/frontend/src/app/expense-table/expense-table.component.ts

```diff
@@ -287,4 +344,46 @@ onTabChange(){
     const blue = parseInt(hexColor.substring(4, 6), 16);
     return `rgba(${red}, ${green}, ${blue}, 0.2)`;
   }
+
+  sortData(sortState:Sort){
+      this.sortBy=sortState.active.toUpperCase();
+      this.ascending=sortState.direction === 'asc' ? true : (sortState.direction === 'desc' ? false : undefined);
+      this.expenseService.getExpensesPage(localStorage.getItem("userId"), this.startDate, this.endDate, this.currentPage, this.pageSize, this.sortBy, this.ascending, this.categoryId, this.currencyId)
```

Comment:
Ca sa nu accesezi in fiecare functie localStorage.getItem("userId"), poti salva userId-ul intr-o variabila in ngOnInit